app/views/handlers/primary_person_list_collector.py

```diff
@@ -31,11 +32,20 @@ def handle_post(self):
             self._primary_person_id = (
                 self.questionnaire_store_updater.add_primary_person(list_name)
             )
+
+            self.questionnaire_store_updater.update_same_name_items(
+                list_name, same_name_answer_ids
+            )
+
             self.evaluate_and_update_section_status_on_list_change(list_name)
             self.questionnaire_store_updater.save()
         else:
             self.questionnaire_store_updater.remove_primary_person(list_name)
 
+            if same_name_answer_ids:
+                self.questionnaire_store_updater.update_same_name_items(
+                    list_name, same_name_answer_ids
+                )
```

Comment:
Any reason to check for `same_name_answer_ids` here, but not above?

Reply:
This was a hangover from get_same_name_answer_ids originally defaulting to False.